index.js

```diff
@@ -522,11 +554,48 @@ class KeyringController extends EventEmitter {
         return { type, data };
       }),
     );
-    const encryptedString = await this.encryptor.encrypt(
-      this.password,
-      serializedKeyrings,
-    );
-    this.store.updateState({ vault: encryptedString });
+
+    let vault;
+    let newExportedKeyString;
+
+    if (this.cacheEncryptionKey) {
+      if (this.password) {
+        const { vault: newVault, exportedKeyString } =
+          await this.encryptor.encryptWithDetail(
+            this.password,
+            serializedKeyrings,
+          );
+
+        vault = newVault;
+        newExportedKeyString = exportedKeyString;
+      } else if (encryptionKey) {
```

Comment:
It looks like this branch isn't tested, because all of the tests have `this.password` set.

Reply:
Fixed!